src/csharp/Grpc.Core/Internal/GrpcThreadPool.cs

```diff
@@ -187,6 +193,16 @@ private void RunHandlerLoop(CompletionQueueSafeHandle cq, IProfiler optionalProf
                 }
             }
             while (ev.type != CompletionQueueEvent.CompletionType.Shutdown);
+
+            // Continuations are running on default threadpool that consists of background threads.
+            // GrpcThreadPool thread (a foreground thread) will not exit unless all queued work had
+            // been finished to prevent terminating the continuations queued prematurely.
+            while (queuedContinuationCounter.Count != 0)
+            {
+                // Only happens on shutdown and having pending continuations shouldn't very common,
+                // so sleeping here for a little bit is fine.
+                Thread.Sleep(FinishContinuationsSleepMillis);
```

Comment:
A little worried about the infinite wait here, maybe max out at something like a 10 second deadline?